test/unit/EchoClientTest.php

```diff
@@ -97,6 +97,88 @@ function testRecentEvents()
         $this->assertEquals(array($expectedEvent),$result);
     }
 
+    function testGetEvents()
+    {
+        $this->setRequiredDefines();
+
+        $stubPersonaClient = $this->getMock('\Talis\Persona\Client\Tokens', array(), array(), '', false);
+        $stubPersonaClient->expects($this->once())->method('obtainNewToken')->will($this->returnValue(array('access_token'=>'some-token')));
+
+        $expectedEvent = array("class"=>"test.expected.event");
+        $response = new \Guzzle\Http\Message\Response('200');
+        $response->setBody(json_encode(array("events"=>array(
+            $expectedEvent
+        ))));
+
+        $mockRequest = $this->getMock('\Guzzle\Http\Message\Request', array('send'), array('get',''));
+        $mockRequest->expects($this->once())->method('send')->will($this->returnValue($response));
+
+        $stubHttpClient = $this->getMock('\Guzzle\Http\Client', array('get'));
+        $stubHttpClient->expects($this->once())->method('get')->with('http://example.com:3002/1/events?limit=25&class=test.expected.event&key=foo&value=bar')->will($this->returnValue($mockRequest));
```

Comment:
Why not use Guzzle mocking here?  http://guzzle3.readthedocs.org/testing/unit-testing.html